### PR TITLE
fix: make message composer less jumpy (AR-912)

### PR DESCRIPTION
**Pull request by trOnk12**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-912" title="AR-912" target="_blank">StoryAR-912</a>  Send text messages
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

The message composer was jumpy on transistion from the state where we display the attachment options to a state when the keyboard is open and other way around. This was caused because of the fact that the UI gets animated on dissapearing of some UI elements, because it has a constrain to align to it. Because of this animation, before the keyboard dissapeared and the attachment options appeared the UI looked to be jumping up and down.

### Solution
I used a static Guideline withtin the ConstraintLayout that is set so that it's offset is equal to the height of the keyboard. This Guideline is used to align MessageComposerInput and AdditionalOptions while we are in either a state where the keyboard is open or when the attachment are displayed, that way we have no constraint on the elements that are animating while the keyboard is being shown or it dissapears. If we have no focus or we are not showing the AdditionalOptions we are aligining the elements as before.

Additional, because there is no way to know the keyboard height before it first pops up on the screen I implemented the logic which gets this value once it appears on the screen, this allows us correctly display the UI with different keyboard sizes, when the keyboard is not shown before the users clicks the attachment options we use a default value. 

jump_fix
change_keyboard_size
f6832fae-2800-4b0a-8284-a78d888d20e6
----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-476/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-476/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-476/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-476/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-476/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-476/7/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt [hunk 1]
```diff
@@ -60,6 +69,9 @@ import com.wire.android.ui.theme.wireColorScheme
 import com.wire.android.ui.theme.wireDimensions
 import com.wire.android.ui.theme.wireTypography
 
+
+private val DEFAULT_KEYBOARD_TOP_SCREEN_OFFSET = 334.dp
```
[the remark this answers is not in the document]
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt:73]: This is how it looks on Pixel 3XL, the action buttons are hidden by the keyboard
Wire 2022-04-06 at 5_15 PM

[the remark this answers is not in the document]
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt:73]: Thanks for the testing i though the actions buttons should jump up when keyboard is shown apperently not :p

**Pull request closed without merging**